src/Bitmap.cpp

```diff
@@ -123,27 +123,10 @@ void OutputBitmap(int tp)
 		OutBaseName = OutFile;
 	}
 
-	switch (tp) {
-		case 0:
-			j = cn1;
-			cn1++;
-			sprintf(OutF, "%s.ge" DIRECTORY_SEPARATOR "%s", OutFile, OutBaseName);
-			break;
-		case 1:
-			j = cn2;
-			cn2++;
-			sprintf(OutF, "%s.ge" DIRECTORY_SEPARATOR "Mean.%s", OutFile, OutBaseName);
-			break;
-		case 2:
-			j = cn3;
-			cn3++;
-			sprintf(OutF, "%s.ge" DIRECTORY_SEPARATOR "Min.%s", OutFile, OutBaseName);
-			break;
-		case 3:
-			j = cn4;
-			cn4++;
-			sprintf(OutF, "%s.ge" DIRECTORY_SEPARATOR "Max.%s", OutFile, OutBaseName);
-			break;
+	for (auto &i : cn) {
+		j = i;
+		i++;
+		sprintf(OutF, "%s.ge" DIRECTORY_SEPARATOR "%s", OutFile, OutBaseName);
 	}
 
```

Comment:
Where have the "Mean, Min, Max" gone from the filenames?

Reply:
I'm also not totally clear on the "for auto" syntax - but don't you basically want something like...

j = cn[tp];
cn[tp]++;
sprintf(OutF, "%s.ge" DIRECTORY_SEPARATOR "%s.%s", OutFile, PreFix[tp], OutBaseName)

where PreFix is an array of {"", "Mean", "Min", "Max"}

Reply:
@weshinsley I don't know what was the intention for the original syntax, I just abstracted it. `for auto` seems the most appropriate for the given situation.

You're absolutely correct about the mean min max part, I missed to automate it. Do you want to implement the syntax (that you proposed) yourself, or want me to implement it instead?

Reply:
In the original code, there's no loop; tp comes in as an argument to the function, and one of the cases in the switch gets called. In your version, you have a loop and all of them will get called won't they?

Reply:
Right, I'll fix the errors you noted.